From 829f38647b7c15dc59ad6b6ebbe8ef4d68a19684 Mon Sep 17 00:00:00 2001
From: Jammy Huang <jammy_huang@aspeedtech.com>
Date: Wed, 18 Mar 2026 15:31:36 +0800
Subject: [PATCH] misc: sli_ast2700: Clear status at SLIM retry

status should be cleared in SLIM retry flow.

Signed-off-by: Jammy Huang <jammy_huang@aspeedtech.com>
Change-Id: Ibfd9dbcf834521dcdd2ad5a62dedbd2fd5db56de
(cherry picked from commit ad3c7e29c75262153dbc800a9c4ca41d9c390980)

Signed-off-by: santhoshs <santhoshs@ami.com>
---
 drivers/misc/sli_ast2700.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/misc/sli_ast2700.c b/drivers/misc/sli_ast2700.c
index 3b48c2fa94..bc349cbeee 100644
--- a/drivers/misc/sli_ast2700.c
+++ b/drivers/misc/sli_ast2700.c
@@ -354,15 +354,11 @@ static int sli_calibrate_mbus_pad_delay(struct sli_data *data, int index, int be
 	int d;
 	int d_first_pass = -1;
 	int d_last_pass = -1;
-	int d_def = 12;
+	int d_def = (begin + end) / 2;
 	int count;
 	char *die_name = (is_k_rx) ? "IOD" : "CPUD";
 	uintptr_t kx = (is_k_rx) ? data->die1.slim : data->die0.slim;
 
-	if (data->die0.phy_clk_freq == SLI_PHYCLK_800M ||
-	    data->die0.phy_clk_freq == SLI_PHYCLK_788M)
-		d_def = 5;
-
 	for (count = 0; count < 50; count++) {
 		for (d = begin; d < end; d++) {
 			sli_set_mbus_delay_single(kx, index, d, is_k_rx);
@@ -387,6 +383,8 @@ static int sli_calibrate_mbus_pad_delay(struct sli_data *data, int index, int be
 		if ((d_last_pass - d_first_pass) >= 3)
 			break;
 		debug("%s SLIM[%d] DS win: {%d, %d} retry %d\n", die_name, index, d_first_pass, d_last_pass, count);
+		d_first_pass = -1;
+		d_last_pass = -1;
 	}
 
 	if (d_first_pass == -1)
@@ -462,6 +460,8 @@ static void sli_calibrate_mbus_delay(struct sli_data *data, bool is_k_rx)
 		if ((d_last_pass - d_first_pass) >= 3)
 			break;
 		debug("%s SLIM DS retry %d\n", die_name, count);
+		d_first_pass = -1;
+		d_last_pass = -1;
 	}
 
 	dc = (d_first_pass + d_last_pass) >> 1;
-- 
2.34.1

